### PR TITLE
feat: #7576 add MultiFilter to styleguidist

### PR DESCRIPTION
**Pull request by jvanveen**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Conventional commits (squash if needed)
- [ ] No warnings during install
- [ ] Updated javascript typing

### CODE REVIEW

**Review by connoratrug — CHANGES_REQUESTED** [review]
The use of  the filter in the example seems to be broken ?

When i type "Apple" nothing happens ( expected options to be filtered)
When i type "Lime" nothing happens ( expected empty option list as Lime is not an option )

**Comment by jvanveen** [comment]
[#7576 Add MultiFilter to components-library](https://trello.com/c/wrcQPFbh/285-7576-add-multifilter-to-components-library)

File: packages/components-library/src/components/filters/MultiFilter.vue [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      get () {
+        return this.value
+      },
+      set (value) {
+        this.$emit('input', value.length === 0 ? undefined : value)
+      }
+    },
+    multifilterOptions () {
+      if (this.inputOptions.length > 0) return this.inputOptions
+      else return this.initialOptions
```
Comment (jvanveen) [inline packages/components-library/src/components/filters/MultiFilter.vue:133]: Dit zorgt ervoor dat de filtering anders werkt dan je zou verwachten

Comment (connoratrug) [reply packages/components-library/src/components/filters/MultiFilter.vue:133]: zou dat niet moeten zijn
 if ( this.inputOptions.length == 0 && queryString == leeg) dan initialOptions anders inputOptions

Comment (connoratrug) [reply packages/components-library/src/components/filters/MultiFilter.vue:133]: fix in https://github.com/molgenis/molgenis-frontend/pull/388/commits/6640ca0d528f52d4358a8aae26e94885c75b1504

**Comment by connoratrug** [comment]
> The use of the filter in the example seems to be broken ?
> 
> When i type "Apple" nothing happens ( expected options to be filtered)
> When i type "Lime" nothing happens ( expected empty option list as Lime is not an option )

fix in https://github.com/molgenis/molgenis-frontend/pull/388/commits/6640ca0d528f52d4358a8aae26e94885c75b1504

File: docker/proxy-config/proxy.d/frontend/lts.conf [hunk 2]
```diff
@@ -10,19 +10,19 @@ location /@molgenis-ui/ {
...
-  rewrite ^/(.*)/search-all/(.*)$             /$1/search-all@0.1.4/$2         break;
-  rewrite ^/(.*)/security/(.*)$               /$1/security@0.2.0/$2           break;
-  rewrite ^/(.*)/settings/(.*)$               /$1/settings@2.0.2/$2           break;
-  rewrite ^/(.*)/data-explorer/(.*)$          /$1/data-explorer@0.35.0/$2     break;
+  rewrite ^/(.*)/questionnaires/(.*)$         /$1/questionnaires@3.0.0/$2     break;
+  rewrite ^/(.*)/scripts/(.*)$                /$1/scripts@3.0.0/$2            break;
+  rewrite ^/(.*)/searchall/(.*)$              /$1/searchall@0.1.5/$2          break;
+  rewrite ^/(.*)/security/(.*)$               /$1/security@0.4.1/$2           break;
+  rewrite ^/(.*)/settings/(.*)$               /$1/settings@4.0.1/$2           break;
+  rewrite ^/(.*)/data-explorer/(.*)$          /$1/data-explorer@0.35.4/$2     break;
```
Comment (jvanveen) [inline docker/proxy-config/proxy.d/frontend/lts.conf:25]: Better create a separate pr for unrelated changes? Or is it related somehow?

File: packages/components-library/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@molgenis-ui/components-library",
-  "version": "0.5.0",
+  "version": "0.6.0",
```
Comment (jvanveen) [inline packages/components-library/package.json:3]: Why is the version bumped here?

File: packages/components-library/src/components/filters/DateTimeFilter.vue [hunk 4]
```diff
@@ -0,0 +1,195 @@
+
```
Comment (jvanveen) [inline packages/components-library/src/components/filters/DateTimeFilter.vue:1]: Wasn't this about MultiFilter?

File: packages/components-library/tests/unit/filters/DateTimeFilter.spec.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
+// @ts-ignore
```
Comment (jvanveen) [inline packages/components-library/tests/unit/filters/DateTimeFilter.spec.ts:1]: Unrelated to MultiFilter

**Review by jvanveen — COMMENTED** [review]
Needs some cleanup

**Comment by connoratrug** [comment]
[#7576 Add MultiFilter to components-library](https://trello.com/c/wrcQPFbh/285-7576-add-multifilter-to-components-library)

**Comment by jvanveen** [comment]
> The use of the filter in the example seems to be broken ?
> 
> When i type "Apple" nothing happens ( expected options to be filtered)
> When i type "Lime" nothing happens ( expected empty option list as Lime is not an option )

When I type "Oran", I get Orange. When I type "Oranz", I get all results. Isn't that unexpected behaviour?

**Review by jvanveen — COMMENTED** [review]
Alleen nog even kijken naar unmatched filter gedrag?

**Review by connoratrug — APPROVED** [review]
